docs/dev_building_web_assembly.rst

```diff
@@ -0,0 +1,146 @@
+
+=========================================
+Building libCellML for use in the Browser
+=========================================
+
+To build libCellML for the browser we use the emscripten toolchain to create web assembly binaries.
+Here we will go through the process of building web assembly binaries on macOS.
+For other operating systems modifications to these instructions will have to be made.
+Any modifications required are expected to be self evident – with this in mind these instructions should be universally applicable.
+
+Preparation
+===========
+
+To build libCellML for the browser we will need the following tools ready for use.
+
+- Emscripten
+- Compiler toolchain
+- CMake (at least version 3.14)
+- Git
+
+Building
+========
+
+libCellML requires two dependencies to be built before libCellML itself can be built.
+We will clone and build those first and then move on to building libCellML.
+
+Dependencies
+------------
+
+Start by creating a directory under which all our files will live and make it our current directory for subsequent commands::
+
+  mkdir emscripten
+  cd emscripten
+
+Now we can clone our dependencies (we suggest using these exact repositories other repositories may work but these repositories are known to work)::
+
+  git clone https://github.com/OpenCMISS-Dependencies/zlib.git -b v1.2.3
+  git clone https://github.com/OpenCMISS-Dependencies/libxml2.git -b v2.9.10
+
+We will create two directories to hold the build files::
+
+  mkdir build-zlib-release
+  mkdir build-libxml2-release
+
+Then we will configure, build, and install zlib as this is required by libxml2.
```

Comment:
Then, we will configure, build, and install zlib as this is required by libxml2.

Reply:
9edec02

Reply:
Not done.

Reply:
760f660